torch/_functorch/_aot_autograd/streams.py

```diff
@@ -20,13 +29,50 @@ def get_stream(node: Node) -> Optional[int]:
         return None
 
 
+def get_stream_or_current_stream(node: Node) -> int:
+    ind = get_stream(node)
+    if ind is None:
+        ind = get_current_stream(get_device(node))
+    return ind
+
+
 def set_stream(node: Node, ind: int) -> None:
     if "custom" in node.meta:
         node.meta["custom"].update({"stream": ind})
     else:
         node.meta["custom"] = {"stream": ind}
 
 
+def insert_sync(
+    graph: torch.fx.Graph,
+    consumer: Node,
+    producer: Node,
+    node_to_wait_event_ind: dict[Node, int],
+) -> None:
+    if producer not in node_to_wait_event_ind:
+        node_to_wait_event_ind[producer] = new_event()
```

Comment:
Are these just stashed as constants on the graph, e.g. if I run the same fx graph again, I will record, and wait on the same event object?

Reply:
good question: dynamo injects bytecode to create these temporary objects on each run. new_event creates an event object for the current graph execution and also signals to dynamo to allocate another event in the prefix bytecode. Next time someone runs the compiled artifact, a new set of objects is created.